torchtune/training/_distributed.py

```diff
@@ -242,7 +245,9 @@ def gather_cpu_state_dict(
         if sharded_param.is_cpu:
             # Move back to device if offloaded to CPU
             sharded_param = sharded_param.to(device)
-        if isinstance(sharded_param._local_tensor, NF4Tensor):
+        if hasattr(sharded_param, "_local_tensor") and isinstance(
+            sharded_param._local_tensor, NF4Tensor
+        ):
```

Comment:
For line 274 (can't comment on that line), what happens if full_tensor() is called on an unsharded parameter? It doesn't crash?

Reply:
This is an edge case, but what if the unsharded param is not on rank0? Will it get moved to cpu properly? What if it's not even on the same node?

Reply:
oh shoot, forgot to copy that logic over from the internal diff